### PR TITLE
Add static analyzer info to DataSet security doc

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
## Summary

@dotpaul has been working on Roslyn static analyzers to accompany the `DataSet` security guidance docs and recently released new versions. We should include a reference to them in the docs.

### CODE REVIEW

**Review by dotpaul — APPROVED** [review]
FWIW, LGTM :smile:

**Comment by GrabYourPitchforks** [comment]
I should confess that Paul did author this section. I don't want to plagiarize his work. :)

File: docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md [hunk 1]
```diff
@@ -482,3 +482,26 @@ Consider replacing the object model to use [Entity Framework](/ef). Entity Frame
 * Offers built-in protections when deserializing data from untrusted sources.
 
 For apps that use `.aspx` SOAP endpoints, consider changing those endpoints to use [WCF](/dotnet/framework/wcf/). WCF is a more fully featured replacement for `.asmx` web services. WCF endpoints [can be exposed via SOAP](../../../wcf/feature-details/how-to-expose-a-contract-to-soap-and-web-clients.md) for compatibility with existing callers.
+
+## Roslyn static code analyzers
+
+Roslyn analyzer security rules, which run as part of compiling source code, can help find vulnerabilities in C# and Visual Basic related to this security issue. Microsoft.CodeAnalysis.FxCopAnalyzers is a NuGet package of Roslyn analyzers, distributed on [nuget.org](https://www.nuget.org/).
```
Comment (gewarren) [inline docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md:488]:
```suggestion
Code analyzer security rules, which run when your source code is compiled, can help to find vulnerabilities related to this security issue in C# and Visual Basic code. Microsoft.CodeAnalysis.FxCopAnalyzers is a NuGet package of code analyzers that's distributed on [nuget.org](https://www.nuget.org/).
```

File: docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md [hunk 2]
```diff
@@ -482,3 +482,26 @@ Consider replacing the object model to use [Entity Framework](/ef). Entity Frame
 * Offers built-in protections when deserializing data from untrusted sources.
 
 For apps that use `.aspx` SOAP endpoints, consider changing those endpoints to use [WCF](/dotnet/framework/wcf/). WCF is a more fully featured replacement for `.asmx` web services. WCF endpoints [can be exposed via SOAP](../../../wcf/feature-details/how-to-expose-a-contract-to-soap-and-web-clients.md) for compatibility with existing callers.
+
+## Roslyn static code analyzers
```
Comment (gewarren) [inline docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md:486]:
```suggestion
## Code analyzers
```
I think we are trying to move away from the use of the term "Roslyn analyzers".

File: docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md [hunk 3]
```diff
@@ -482,3 +482,26 @@ Consider replacing the object model to use [Entity Framework](/ef). Entity Frame
...
+
+Roslyn analyzer security rules, which run as part of compiling source code, can help find vulnerabilities in C# and Visual Basic related to this security issue. Microsoft.CodeAnalysis.FxCopAnalyzers is a NuGet package of Roslyn analyzers, distributed on [nuget.org](https://www.nuget.org/).
+
+For an overview of Roslyn analyzers, see [Overview of source code analyzers](https://docs.microsoft.com/visualstudio/code-quality/roslyn-analyzers-overview).
```
Comment (gewarren) [inline docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md:490]:
```suggestion
For an overview of code analyzers, see [Overview of source code analyzers](https://docs.microsoft.com/visualstudio/code-quality/roslyn-analyzers-overview).
```

File: docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md [hunk 4]
```diff
@@ -482,3 +482,26 @@ Consider replacing the object model to use [Entity Framework](/ef). Entity Frame
...
+
+The new Microsoft.CodeAnalysis.FxCopAnalyzers rules to enable are:
```
Comment (gewarren) [inline docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md:492]:
```suggestion
Enable the following Microsoft.CodeAnalysis.FxCopAnalyzers rules:

```

File: docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md [hunk 5]
```diff
@@ -482,3 +482,26 @@ Consider replacing the object model to use [Entity Framework](/ef). Entity Frame
...
+- [CA2353](https://docs.microsoft.com/visualstudio/code-quality/ca2353): Unsafe DataSet or DataTable in serializable type
+- [CA2354](https://docs.microsoft.com/visualstudio/code-quality/ca2354): Unsafe DataSet or DataTable in deserialized object graph can be vulnerable to remote code execution attacks
+- [CA2355](https://docs.microsoft.com/visualstudio/code-quality/ca2355): Unsafe DataSet or DataTable type found in deserializable object graph
+- [CA2356](https://docs.microsoft.com/visualstudio/code-quality/ca2356): Unsafe DataSet or DataTable type in web deserializable object graph
+- [CA2361](https://docs.microsoft.com/visualstudio/code-quality/ca2361): Ensure autogenerated class containing DataSet.ReadXml() is not used with untrusted data
+- [CA2362](https://docs.microsoft.com/visualstudio/code-quality/ca2362): Unsafe DataSet or DataTable in autogenerated serializable type can be vulnerable to remote code execution attacks
+
+For more information on configuring rules, see [Use rule sets to group code analysis rules](https://docs.microsoft.com/visualstudio/code-quality/using-rule-sets-to-group-code-analysis-rules).
```
Comment (gewarren) [inline docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md:503]: Rule sets are deprecated in favor of EditorConfig files.

```suggestion
For more information about configuring rules, see [Use Use code analyzers](https://docs.microsoft.com/visualstudio/code-quality/use-roslyn-analyzers).
```

File: docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md [hunk 6]
```diff
@@ -482,3 +482,26 @@ Consider replacing the object model to use [Entity Framework](/ef). Entity Frame
...
+
+These new rules are available in:
```
Comment (gewarren) [inline docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md:505]:
```suggestion
The new security rules are available in the following NuGet packages:

```

File: docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md (hunk 5 above)
Comment (GrabYourPitchforks) [reply docs/framework/data/adonet/dataset-datatable-dataview/security-guidance.md]: Suggestion had a typo, fixed in the latest PR iteration.

**Comment by GrabYourPitchforks** [comment]
@gewarren I merged your feedback, thank you!

**Review by gewarren — APPROVED** [review]

**Pull request merged**